From 6b0b687638fa6707eacab59b8e529a87fcd169f4 Mon Sep 17 00:00:00 2001
From: Darren Etheridge <detheridge@ti.com>
Date: Wed, 5 Feb 2014 11:41:22 -0600
Subject: [PATCH 20/71] drm/tilcdc: restore correct display mode and contents
 on pm resume

This commit comes from branch ti-linux-3.12.y:
  git://git.ti.com/ti-linux-kernel/ti-linux-kernel.git

On resume the screen contents were not being restored properly.  Looking at
other DRM drivers it seems a call to drm_helper_resume_force_mode() is needed
in the resume handler to force restoration of the mode and framebuffer data.

Signed-off-by: Darren Etheridge <detheridge@ti.com>
(cherry picked from commit 93350dbe304b5c2aba5225c20c827295a74963ea)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/tilcdc/tilcdc_drv.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/tilcdc/tilcdc_drv.c b/drivers/gpu/drm/tilcdc/tilcdc_drv.c
index 8122a31..27aa343 100644
--- a/drivers/gpu/drm/tilcdc/tilcdc_drv.c
+++ b/drivers/gpu/drm/tilcdc/tilcdc_drv.c
@@ -572,6 +572,13 @@ static int tilcdc_pm_resume(struct device *dev)
 		if (registers[i].save && (priv->rev >= registers[i].rev))
 			tilcdc_write(ddev, registers[i].reg, priv->saved_register[n++]);
 
+	/*
+	 * if this call isn't here, the display is blank on return from
+	 * suspend.  With this call here the contents of the framebuffer
+	 * during suspend are restored correctly.
+	 */
+	drm_helper_resume_force_mode(ddev);
+
 	drm_kms_helper_poll_enable(ddev);
 
 	return 0;
-- 
1.7.5.4

